web/packages/teleport/src/Roles/RoleEditor/StandardEditor/sections.tsx

```diff
@@ -66,17 +67,56 @@ export const SectionBox = ({
   validation?: ValidationResult;
   onRemove?(): void;
 }>) => {
+  enum ExpansionState {
+    /** The section is fully collapsed. */
+    Collapsed,
+    /** The section is fully expanded. */
+    Expanded,
+    /**
+     * The section is still expanded, but it's showing a collapsing animation.
+     * In this state, the <details> element is still open to make all its
+     * children visible.
+     */
+    Collapsing,
+  }
+
   const theme = useTheme();
-  const [expanded, setExpanded] = useState(true);
-  const ExpandIcon = expanded ? Minus : Plus;
-  const expandTooltip = expanded ? 'Collapse' : 'Expand';
+  const [expansionState, setExpansionState] = useState(ExpansionState.Expanded);
```

Comment:
For some reason, on Safari the animation works only for collapsing the section. 
But it doesn't seem that we can do much about it, it's fine in other browsers.

https://github.com/user-attachments/assets/0d7a5b4d-132e-4c6e-ac9e-35b3232b8f7f

Reply:
Interesting. I'll take a look. The initial code actually included one more state, dedicated solely to measuring the height of the component, but then it somehow was unnecessary, and I think I forgot to retest on Safari after I simplified it.